jax/experimental/shard_map.py

```diff
@@ -942,7 +944,7 @@ def to_val_rep_pair(self, val):
       raise Exception(f"Shouldn't have any non-shard_map tracers: {val}")
     else:
       val_ = _unmatch_spec(self.mesh, {}, val, self.context_mesh)
-      return val_, None
+      return val_, frozenset(self.mesh.axis_names)
```

Comment:
How about `frozenset(self.mesh.axis_names) - auto`? Would have to pass in `auto` from the caller(s). (I hope they have it...)

Reply:
Or maybe `frozenset(self.context_mesh.manual_axes)`

Reply:
Yes! Awesome ❤️ 

But this probably: `frozenset(for n in self.context_mesh.axis_names if self.context_mesh._name_to_type[n] == AxisType.Manual)`

I can expose a `.manual_axes` property on AbstractMesh and we can replace this with that. Justin you can do that too if you want here: https://cs.opensource.google/jax/jax/+/main:jax/_src/mesh.py;l=176?q=AbstractMesh&ss=jax%2Fjax in `_BaseMesh`

Reply:
Changing to frozenset(self.mesh.axis_names)

Reply:
Yeah sorry for misleading you and Matt.

`context_mesh` is the mesh in context outside of shmap. So it cannot have any Manual axes but it can contain a combo of auto and explicit axes which `self.mesh` doesn't have.